### PR TITLE
Developed version 2.0.5

### PR DESCRIPTION
**Pull request by MichaelRoeder**
Hotfixes for #201, #273, #315 and #304.

### CODE REVIEW

File: platform-controller/src/main/java/org/hobbit/controller/PlatformController.java [hunk 1]
```diff
@@ -641,9 +719,19 @@ public void handleFrontEndCmd(byte bytes[], String replyTo, BasicProperties repl
...
+     * selecting a certain graph if the graphUri is {@code null}).
+     * 
+     * @param challengeUri
+     *            the URI for which the model should be retrieved
+     * @param graphUri
+     *            the URI from which the data should be retrieved or {@code null} if
+     *            all graphs should be taken into account.
+     * @return the RDF model of the challenge
+     */
+    protected Model getChallengeFromUri(String challengeUri, String graphUri) {
```
Comment (earthquakesan) [inline platform-controller/src/main/java/org/hobbit/controller/PlatformController.java]: Is it safe to discard one argument method in favor of two argument method?

File: platform-controller/src/main/java/org/hobbit/controller/tools/ProjectContentChecker.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.hobbit.controller.gitlab.GitlabController;
+import org.hobbit.controller.gitlab.GitlabControllerImpl;
+import org.hobbit.controller.gitlab.Project;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProjectContentChecker {
```
Comment (earthquakesan) [inline platform-controller/src/main/java/org/hobbit/controller/tools/ProjectContentChecker.java]: This class is not covered with unit tests. Where is it used?

File: platform-controller/src/main/java/org/hobbit/controller/tools/ProjectSearcher.java [hunk 3]
```diff
@@ -16,6 +16,7 @@ public static void main(String[] args) throws InterruptedException {
         if (args.length < 2) {
             System.out.println(
                     "Error: wrong usage. The following parameters are necessary:\n <security-token> <search-term(s)>");
```
Comment (earthquakesan) [inline platform-controller/src/main/java/org/hobbit/controller/tools/ProjectSearcher.java]: Actually, the whole tools package is not covered. Can it be fixed?

File: platform-controller/src/main/java/org/hobbit/controller/PlatformController.java (hunk 1 above)
Comment (MichaelRoeder) [reply platform-controller/src/main/java/org/hobbit/controller/PlatformController.java]: What exactly do you mean with safe?

The reason for this change is that the original method was only requesting information from the challenge definition graph. It was used in two different methods of the platform. 
However, it turned out that one of them has to query all graphs while the other has to query only the challenge def. graph. This solution makes sure that there is still only one method for retrieving a challenge model which can be used in both situations.

File: platform-controller/src/main/java/org/hobbit/controller/tools/ProjectContentChecker.java (hunk 2 above)
Comment (MichaelRoeder) [reply platform-controller/src/main/java/org/hobbit/controller/tools/ProjectContentChecker.java]: Everyt class in the `tool` package is a simple cmd line tool that can be executed to check projects or search for a certain project or something similar. It is no functionality that is necessary for the platform but it is helpful for the administration or development.

I will add comments to all classes to emphasize their functionality. However, I wouldn't invest time into writing JUnit tests for them as long as the functionality is not needed for the rest of the platform.

File: platform-controller/src/main/java/org/hobbit/controller/tools/ProjectSearcher.java (hunk 3 above)
Comment (MichaelRoeder) [reply platform-controller/src/main/java/org/hobbit/controller/tools/ProjectSearcher.java]: See comment above.

**Comment by earthquakesan** [comment]
I've checked interface according to Micha:
```
You have defined a challenge with a task. If you want to edit the task, the values are either not shown at all or you see the default values. But you do not see the values that you defined before, although they are stored correctly.
```

Other than that fields can receive any kind of values (for example, I can put string into int field and will not get any error).

**Comment by MichaelRoeder** [comment]
The issue above has been documented with #322.

**Pull request merged**